### PR TITLE
FortiNDR Cloud: New App - Submitting new app for review

### PR DESCRIPTION
**Pull request by FortiNDR-Integration**
Please ensure your pull request (PR) adheres to the following guidelines:

- Please refer to our contributing documentation for any questions on submitting a pull request, link: [Contribution Guide](https://github.com/splunk-soar-connectors/.github/blob/main/.github/CONTRIBUTING.md)

#### Please check if your PR fulfills the following requirements:
- [ x] Testing of all the changes has been performed (for bug fixes / features)
- [x ] The manual_readme_content.md has been reviewed and added / updated if needed (for bug fixes / features)
- [ x] Use the following format for the PR description: `<App Name>: <PR Type> - <PR Description>`
- [ x] Provide release notes as part of the PR submission which describe high level points about the changes for the upcoming GA release.
- [ ] Verify all checks are passing.
- [ x] Do NOT use the `next` branch of the forked repo. Create separate feature branch for raising the PR.
- [ x] Do NOT submit updates to dependencies unless it fixes an issue.
#### Please check the type of change your PR introduces:
- [ x] New App
- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Documentation
- [ ] Other (please describe):

## Security Considerations (REQUIRED)
- [ ] If you are exposing any endpoints using a [REST handler](https://docs.splunk.com/Documentation/SOAR/current/DevelopApps/RESTHandlers),
  please document them in the `manual_readme_content.md`.
- [ ] If this is a new connector or you are adding new actions
    - [ x] Please document in the `manual_readme_content.md` all methods (eg, OAuth) used to authenticate
      with the service that the connector is integrating with.
    - [ ] If any actions are unable to run on SOAR Cloud, please document this in the `manual_readme_content.md`.
- [ ] Are you introducing any new cryptography modules? If yes, please elaborate their purpose:
- [ ] Are you are accessing the file system? If yes, please verify that you are only accessing paths returned through
the [Vault](https://docs.splunk.com/Documentation/SOARonprem/5.2.1/DevelopApps/AppDevAPIRef#Vault) API.
- [ ] Are you are marking code to be ignored by Semgrep with [`nosemgrep`](https://semgrep.dev/docs/ignoring-files-folders-code/#ignoring-code-through-nosemgrep)?
If yes, please provide justification in an additional comment next to the ignored code.

## Release Notes (REQUIRED)
- First release.

## What is the current behavior? (OPTIONAL)
- Describe the current behavior that you are modifying.

## What is the new behavior? (OPTIONAL)
- Describe the behavior or changes that are being added by this PR.

## Other information (OPTIONAL)
- Any other information that is important to this PR such as screenshots of how the component looks before and after the change.

## Pay close attention to (OPTIONAL)
- Any specific code change or test case points which must be addressed/reviewed at the time of GA release.

## Screenshots (if relevant)

---
Thanks for contributing!

### CODE REVIEW

**Comment by ishans-crest** [comment]
Hi @Fnc-integration ,

We have made changes according to our coding standards. There are some suggestions/queries from our side:

- Would it be fine if we change `min_phantom_version` to `6.0.2` in the app's JSON file as per our standard best practice?

- All the action's output doesn't need a `column_name` key. To display a specific key as a column in the output, you should use the `column_name` paired with the `column_order` key. You can refer below example

`{
"data_path": "action_result.data.*.tasks.*.actual_start_time",
                    "data_type": "string",
                    "example_values": [
                        "2019-01-30T00:00:00.000Z"
                    ],
                    "column_name": "Start Time",
"column_order": 0
}`

The column will be shown according to the ascending `column_order` provided by you(Here zero means the first column).

- All the action's output required `action_result.parameter` data paths.

**Comment by FortiNDR-Integration** [comment]
Hi @ishans-crest

I updated the minimum version to 6.0.2 and added the action_result.parameter to all the actions.
I have a question regarding the column_name and the column_order fields. Will all the output fields be shown regardless if they have a column name?

Also, i have some trouble fixing some of the PR's static checks. I fixed most of them but there are three that I am not sure how to fix. The comment on the report are not to clear.

- **FAILED - repo_name_has_expected_app_id**: It sais to add the app id for fortindrcloud to data/repo_name_to_appid.json but I do not know where that json file is.
- **FAILED - document_commonly_used_service_ports**: I do not think it need any specific port opened on Phantom since it communicate with our service using a REST API
- **FAILED - app_actions - Some action names were unknown**: I am not sure what to do for this one.

I will appreciate any guidance to fix these three last failing checks.

**Comment by ishans-crest** [comment]
Hi @Fnc-integration ,

The output fields will be displayed on the Splunk SOAR platform's UI only if both the `column_name` and `column_order` fields are present; otherwise, the UI will show an empty output.

- While reviewing the code I noticed that you've provided the `action_result.parameter` data path in all the actions. It's important to ensure that this path contains the action parameter specified within each respective action.

For Example,
```
parameters": {
  "account_uuid": {
	  "description": "UUID of account to filter by",
	  "data_type": "string",
           "order": 0
	},
	output": [
               {
	            "data_path": "action_result.parameter.account_uuid",
	            "data_type": "string"
	       },
               ..........
               ......
               ...
          ]
}
```
The output data path will look like the above for all the parameters provided in action.

As for the static checks, those will be done from our side.

**Comment by FortiNDR-Integration** [comment]
Hi @ishans-crest
I updated the action_result.parameter as requested and added the colum_order field.
By the way, I noticed that after the change made on the dependencies I was getting an error "Failed to import the python connector module." on my local dev env. The logs shows it failed importing the dateparser dependency. 

Also, could you explain me what is the process for publishing the app after the PR is approved? I would like to have an idea of how long it takes for the app to be published.

Thank

**Comment by ishans-crest** [comment]
Hi @Fnc-integration ,
I've resolved the importing issue. Also while reviewing the PR I noticed that the `api_base_url` configuration parameter was missing in JSON. This parameter is used in the 'initialize' block of connectors code. I went through your old commits and added this configuration parameter. Could you please confirm this with the latest build and asset?

It generally takes a couple of weeks to get everything approved and the new app to be listed on Splunkbase.

**Comment by FortiNDR-Integration** [comment]
Hi @ishans-crest

The api_base_url config argument was for a functionality that we decided not to add yet on this first version. I removed the argument and its use.

What is the next step in the process? Do I need to do anything else?

**Comment by ishans-crest** [comment]
@Fnc-integration Thanks for the clarification regarding the `api_base_url` parameter. Once the static tests are resolved from our side. we will proceed with the merging and release process.
>Do I need to do anything else?

We will let you know if we require your help.

**Comment by nastorSplunk** [comment]
@ishans-crest to verify, we are going to take this and own the process to getting it to Splunkbase, correct?

**Comment by FortiNDR-Integration** [comment]
Hi @ishans-crest, @achandya-crest, @mishalp-crest,  @aparekh-crest,
What is the status of this PR? It has been on for a month now. Can we get an eta of it to be reviewed?
Thank

**Review by achandya-crest — APPROVED** [review]

File: fortindrcloud.json [hunk 1]
```diff
@@ -0,0 +1,4063 @@
...
+    "configuration": {
+        "api_key": {
+            "name": "api_key",
+            "data_type": "password",
+            "required": true,
+            "description": "API Token to connect to FortiNDR Cloud RESTful APIs",
+            "order": 0
+        },
+        "ph": {
+            "data_type": "ph",
```
Comment (splunk-jessica) [inline fortindrcloud.json:34]: Remove unused config param

File: fortindrcloud.json [hunk 2]
```diff
@@ -0,0 +1,4063 @@
...
+            "order": 1
+        },
```
Comment (splunk-jessica) [inline fortindrcloud.json:36]:
```suggestion
```
remove unused configuration param

Comment (splunk-jessica) [reply fortindrcloud.json:36]: Will likely have to update the `"order"` keys on all other params here

**Review by splunk-jessica — APPROVED** [review]

**Review by ishans-crest — APPROVED** [review]

**Pull request merged**